addons/mrp/models/mrp_workcenter.py

```diff
@@ -154,9 +155,26 @@ def _compute_productive_time(self):
 
     @api.depends('blocked_time', 'productive_time')
     def _compute_oee(self):
+        time_data = self.env['mrp.workcenter.productivity'].read_group(
+            domain=[
+                ('date_start', '>=', fields.Datetime.to_string(datetime.now() - relativedelta.relativedelta(months=1))),
+                ('workcenter_id', 'in', self.ids),
+                ('date_end', '!=', False),
+            ],
+            fields=['workcenter_id', 'loss_type', 'duration'],
+            groupby=['workcenter_id', 'loss_type'],
+            lazy=False,
+        )
+        time_by_order = defaultdict(lambda: {'productive_time': 0.0, 'blocked_time': 0.0})
+        for td in time_data:
+            time_to_update = 'productive_time' if td['loss_type'] == 'productive' else 'blocked_time'
+            time_by_order[td['workcenter_id'][0]][time_to_update] += td['duration']
         for order in self:
```

Comment:
```suggestion
        for workcenter in self:
```
Pet peeve of mine, but since we're rewriting the compute anyway... Using `order` used here is misleading.